test/box/gh-6934-nonprintable-error-message.result

```diff
@@ -0,0 +1,5 @@
+box.error.new(box.error.ILLEGAL_PARAMS, 'bad: \x80, ok: 🕷')
```

Comment:
Please, use luatest for new tests.

Reply:
Can we call `:hex()` on the strings?

Reply:
> Can we call `:hex()` on the strings?

```lua
t.assert_equals('xxx', box.error.new(box.error.ILLEGAL_PARAMS, string.fromhex('80')))
```
leads to the same error, and this too:
```lua
t.assert_equals('xxx', box.error.new(box.error.ILLEGAL_PARAMS, string.format("%c", 0x80)))
```

Reply:
The key problem is that test-run can't proceed with non-text test output (when it works under Python 3). It was deliberate choice: we can support such cases in test-run (it costs some code complexity), but they're rare and easy to change to output ascii/utf-8.

So the goal is to change the TAP13 output from luatest to adhere utf-8. luatest outputs actual and expected values in case of an error and we should make it utf-8 compliant. So my proposal is the following:

```
-- Use :hex(), because bla bla.
local err_msg = tostring(box.error.new(box.error.ILLEGAL_PARAMS, '\x80')
t.assert_equals(('xxx'):hex(), err_msg:hex())
```

Reply:
Switched to luatest.